packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1283,7 +1283,10 @@ export class MergeTree {
 			);
 		}
 
-		if (searchInfo.tile) {
+		if (
```

Comment:
I think this only fixes the immediate issue filed, but the documentation of this method suggests the position being searched should be exclusive w.r.t the search. For example, if searching at index 4 in `"abcd<marker>efg"`, we don't get the marker in either direction as it's at the exact position.

Could you add tests to cover this case? I expect something like this change will need to be made to `recordTileStart` and `tileShift`, where you'll need to make use of `segpos` and/or `offset` (will need to look at semantics of those)

Reply:
Ah, I'd assumed it was a problem with 0 and length specifically. If it applies to positions in-between, that could be a breaking change to existing usages. Perhaps it would be better to update the documentation to indicate the behaviour is inclusive.

Reply:
I think it might just be a problem with 0/length - I added a test case to cover the scenario above, and findTile in the middle of the string seems to be exclusive. It returns undefined for both the forward and backward searches without any changes made to recordTileStart or tileShift.

Reply:
I had a closer look at the code, I think it might be buggy for markers at the start (and maybe end) of blocks as well. Do you still get the same results if you construct a similar setup to above but with "abcd" in one block and "<marker>efg" in another? (will need to vary exact text to make that happen)

Reply:
mostly being a stickler as this fix seems really odd to me. i'd guess we have further issues. I'd be more confident it was correct if we did something like add a combinatorial test which compared block-accelerated search to forward+backward excursions for a semi-exhaustive list of cases

Reply:
> Do you still get the same results if you construct a similar setup to above but with "abcd" in one block and "efg" in another?

Yep, findTile still returns undefined with the separated blocks. 

There's definitely something wrong with the fix that I have now though, since one of the SharedText jest tests is failing, and it looks like findTile does get called at some point during the execution. I haven't looked into it much yet, I'm just starting to now.